src/main/java/com/github/manolo8/darkbot/extensions/plugins/IssueHandler.java

```diff
@@ -25,6 +28,56 @@ public IssueHandler(IssueHandler parent) {
         this.listener.add(uiListener::send);
     }
 
+    public static void handleLoadFeatureException(FeatureDefinition<?> fd, Throwable e) {
+        handleFeatureException(fd, PluginIssue.Level.ERROR, "bot.issue.feature.failed_to_load", e);
+    }
+
+    public static boolean handleTickFeatureException(FeatureDefinition<?> fd, PluginIssue.Level defaultLevel, Throwable e) {
+        return handleFeatureException(fd, defaultLevel, "bot.issue.feature.failed_to_tick", e);
+    }
+
+    /**
+     * @return true if exception was {@link FeatureException#isCritical()}, false otherwise
+     */
+    private static boolean handleFeatureException(FeatureDefinition<?> fd, PluginIssue.Level defaultLevel,
+                                                  String defaultMessage, Throwable e) {
+        Objects.requireNonNull(fd, "Feature Definition cannot be null!");
+
+        FeatureException featureException = FeatureException.find(e);
+        if (featureException != null) {
+            fd.getIssues().addIssue(featureException);
+            return featureException.isCritical();
+        }
+
+        PluginIssue issue = new PluginIssue(defaultMessage, IssueHandler.createDescription(e), defaultLevel, e);
+        fd.getIssues().addIssue(issue);
+        return false;
+    }
+
+    private void addIssue(FeatureException fe) {
+        String message = fe.getMessage();
+        String description = fe.getDescription() == null ? createDescription(fe.getCause()) : fe.getDescription();
+        PluginIssue.Level level = PluginIssue.Level.values()[fe.getLevel().ordinal()];
+
+        PluginIssue issue = new PluginIssue(message, description, level, fe);
+        addIssue(issue);
+    }
+
+    private void addIssue(PluginIssue pluginIssue) {
```

Comment:
a method called `add(PluginIssue pluginIssue)` already exists, which does basically this exact functionality (add it, and send an update to the listener if it's new)

You should update that method instead of creating a copy-cat one, as that method should also do increase and print logic